File: 3-Queue/Java/Main.java
/**
 * Main - a simple test of the queue data structure
 */
public class Main {

	public static void main(String argv[]) {

		// Declare an object of type QueueX
		QueueX qX = new QueueX(10);

		// pushing the values to the queue
		qX.Enqueue(10);
		qX.Enqueue(9);
		qX.Enqueue(8);
		qX.Enqueue(7);
		qX.Enqueue(6);
		qX.Enqueue(5);
		qX.Enqueue(4);
		qX.Enqueue(3);
		qX.Enqueue(2);
		qX.Enqueue(1);

		// prints the value at peek it must be 10
		System.out.println("peek = " + qX.Peek() + "\n");

		// Prints the queue
		qX.Print();

		// removing using Dequeue method
		System.out.println("\nvalue removed = " + qX.Dequeue());
		System.out.println("value removed = " + qX.Dequeue());
		System.out.println("value removed = " + qX.Dequeue());
		System.out.println("value removed = " + qX.Dequeue());
		System.out.println("value removed = " + qX.Dequeue());

		// removing using Delete method
		System.out.println("\nvalue removed = " + qX.Delete());
		System.out.println("value removed = " + qX.Delete());
		System.out.println("value removed = " + qX.Delete());
		System.out.println("value removed = " + qX.Delete());
		System.out.println("value removed = " + qX.Delete());

		// checking if the queue is empty or not
		if (qX.isEmpty()) {

			System.out.println("\ncongratulations - The queue is empty.");
		}
		else {

			System.out.println("\nYou fucked up. check your code it isn't working.");
		}
	}
}

File: 3-Queue/cpp/main.cpp
#include "queue.h"
#include <iostream>

/**
* main - A simple test of the queue data structure
* Return: Always 0
*/

int main(int argc, char **argv)
{
	QueueClass Queue = QueueClass(10);
	/* Push to the queue*/
	Queue.Push(10);
	Queue.Push(9);
	Queue.Push(8);
	Queue.Push(7);
	Queue.Push(6);
	Queue.Push(5);
	Queue.Push(4);
	Queue.Push(3);
	Queue.Push(2);
	Queue.Push(1);

	/* Check for the error handling*/
	Queue.Push(100);

	/* Just a new line for visibility*/
	std::cout << std::endl;

	/* Get the element at the peek of the queue*/
	std::cout << "Peek = " << Queue.Peek() << std::endl;


	/* Just a new line for visibility*/
	std::cout << std::endl;

	/* Print the queue*/
	Queue.Print();

	/* Just a new line for visibility*/
	std::cout << std::endl;

	/* removes all the element of the queue*/
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;
	std::cout << "element removed = " << Queue.Pop() << std::endl;

	/* Just a new line for visibility*/
	std::cout << std::endl;

	/* Check for error handling*/
	std::cout << "Peek = " << Queue.Peek() << std::endl;

	/* Just a new line for visibility*/
	std::cout << std::endl;

	/* Check if the queue is empty or not*/
	if (Queue.isEmpty())
	{
		std::cout << "Congratulations - The Queue is empty and your code is " 
			<< "working" << std::endl;
	}
	else 
	{
		std::cout << "You fucked up. Your code isn't working." << std::endl;
	}

	return (0);
}
